core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -45,6 +45,7 @@ final class TableScanContext {
   private final ExecutorService planExecutor;
   private final boolean fromSnapshotInclusive;
   private final MetricsReporter metricsReporter;
+  private final String ref;
```

Comment:
I think we should be explicit here and call it `branch`, this ref should always be a branch.